src/components/PeopleTable.tsx

```diff
@@ -1,5 +1,127 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-export const PeopleTable = () => {
+import { useSearchParams } from 'react-router-dom';
+import cn from 'classnames';
+import {
+  Dispatch,
+  SetStateAction,
+  useCallback,
+  useEffect,
+  useMemo,
+} from 'react';
+import { PersonLink } from './PersonLInk';
+import { SearchParams } from '../types/SearchParams';
+import { Person } from '../types';
+import { SearchLink } from './SearchLink';
+
+type Props = {
+  people: Person[];
+  filteredPeople: Person[];
+  setFilteredPeople: Dispatch<SetStateAction<Person[]>>;
+};
+
+export const PeopleTable: React.FC<Props> = ({
+  people,
+  filteredPeople,
+  setFilteredPeople,
+}) => {
+  const [searchParams] = useSearchParams();
+
+  const sortParam = searchParams.get('sort') || null;
+  const orderParam = searchParams.get('order') || null;
+  const query = searchParams.get('query') || '';
+  const sex = searchParams.get('sex') || '';
+  const century = useMemo(
+    () => searchParams.getAll('centuries') || [],
+    [searchParams],
+  );
+
+  const getIconClass = (param: string): string => {
+    if (sortParam !== param) {
+      return 'fa-sort';
+    }
+
+    if (sortParam === param && !orderParam) {
+      return 'fa-sort-up';
+    }
+
+    if (sortParam === param && orderParam === 'desc') {
+      return 'fa-sort-down';
+    }
+
+    return '';
+  };
+
+  const filterPeople = useCallback(() => {
```

Comment:
```suggestion
  const filteredPeople = useMemo(() => {
```
And instead of `setFilteredPeople(filtered);` return `filtered`;

filteredPeople state is redundant - delete it